File: C++/Medium/Accepted/execution-of-all-suffix-instructions-staying-in-a-grid_3.cpp
class Solution {
public:
    vector<int> executeInstructions(int n, vector<int>& startPos, string s) {
        
        vector<int> result;
        int m = s.size();
        for (int i = 0; i < m; ++i) {
            int x = startPos[0], y = startPos[1];
            int count = 0;
            for (int j = i; j < m; ++j) {
                if (s[j] == 'L' && y > 0) {
                    --y;
                    ++count;
                } else if (s[j] == 'R' && y < n - 1) {
                    ++y;
                    ++count;
                } else if (s[j] == 'U' && x > 0) {
                    --x;
                    ++count;
                } else if (s[j] == 'D' && x < n - 1) {
                    ++x;
                    ++count;
                } else {
                    break;
                }
            }
            result.push_back(count);
        }
        return result;
    }
};

File: Rust/Medium/CompileError/execution-of-all-suffix-instructions-staying-in-a-grid_3.rs
impl Solution {
    pub fn execute_instructions(n: i32, start_pos: Vec<i32>, s: String) -> Vec<i32> {
        
        let mut res = vec![0; s.len()];
        let mut pos = start_pos;
        let mut dir = vec![(0, 1), (1, 0), (0, -1), (-1, 0)];
        let mut cur_dir = 0;
        for (i, c) in s.chars().enumerate() {
            let (dx, dy) = dir[cur_dir];
            match c {
                'L' => cur_dir = (cur_dir + 3) % 4,
                'R' => cur_dir = (cur_dir + 1) % 4,
                'U' => {
                    if pos[0] + dx >= 0 {
                        pos[0] += dx;
                    }
                }
                'D' => {
                    if pos[0] + dx < n {
                        pos[0] += dx;
                    }
                }
                _ => {}
            }
            res[i] = (pos[0] - start_pos[0]).abs() + (pos[1] - start_pos[1]).abs();
        }
        res
    }
}